ipa-core/src/cli/playbook/mod.rs

```diff
@@ -106,18 +113,41 @@ pub fn validate_dp(expected: Vec<u32>, actual: Vec<u32>, epsilon: f64, per_user_
 
         let next_expected_f64: f64 = next_expected.unwrap().into();
         let actual_expect_f64: f64 = next_actual.unwrap().into();
+
         let noise_params = NoiseParams {
             epsilon,
+            per_user_credit_cap,
             ell_1_sensitivity: per_user_credit_cap.into(),
             ell_2_sensitivity: per_user_credit_cap.into(),
             ell_infty_sensitivity: per_user_credit_cap.into(),
             dimensions: 256.0, // matches the hard coded number of breakdown keys in oprf_ipa.rs/execute
             ..Default::default()
         };
-
-        let (mean, std) = crate::protocol::dp::noise_mean_std(&noise_params);
-        let same = actual_expect_f64 - mean > next_expected_f64 - 10.0 * std
-            && actual_expect_f64 - mean < next_expected_f64 + 10.0 * std;
+        let same = match dp_mechanism {
+            DpMechanism::Binomial { epsilon: _ } => {
+                let (mean, std) = crate::protocol::dp::binomial_noise_mean_std(&noise_params);
+                actual_expect_f64 - mean > next_expected_f64 - 10.0 * std
+                    && actual_expect_f64 - mean < next_expected_f64 + 10.0 * std
+            }
+            DpMechanism::DiscreteLaplace { epsilon: _ } => {
+                let truncated_discrete_laplace = OPRFPaddingDp::new(
+                    noise_params.epsilon,
+                    noise_params.delta,
+                    noise_params.per_user_credit_cap,
+                )
+                .unwrap();
+
+                let (_, std) = truncated_discrete_laplace.mean_and_std();
+                let tolerance_factor = 12.0;
+                // println!("mean = {mean}, std = {std}, tolerance_factor * std = {}",tolerance_factor * std);
+                (actual_expect_f64 - next_expected_f64).abs() < tolerance_factor * std
+            }
+            DpMechanism::NoDp => {
+                let tolerance = 0.001;
```

Comment:
why isn't this tolerance 0?

Reply:
we are doing a float comparison here, so we have some tolerance to account for small round-off errors (I think it was a clippy suggestion to add in case of float comparisons).  But let me see if we can use the earlier u32 representations here instead.

Reply:
okay updated to change this.